eppic-cli/src/main/java/eppic/assembly/Assembly.java

```diff
@@ -955,21 +974,64 @@ public String toString() {
 	}
 
 	/**
-	 * Assign a call to this Assembly, retrieve the call subsequently with {@link #getCall()}
+	 * Compute an initial (non-normalized) probabilistic score from
+	 * the individual interface probabilities.
 	 */
-	public void score() {
-
-		// we won't support non-fully-covering stoichiometries for the moment
-		// TODO support them: most likely requires a more complex data model where we can have score/calls for each of the subcomponents of the assembly
-		if (!assemblyGraph.isFullyCovering()) {
-			logger.warn("Assembly {} does not cover all entities, assembly scoring will be done for first sub-assembly only", toString());
-		}
+	public void calcScore() {
 		
+		// Construct a list of all the smaller powersets of this assembly that are equivalent
+		List<PowerSet> pss = new ArrayList<PowerSet>();
 		
-		SubAssembly firstSubAssembly = assemblyGraph.getSubAssemblies().get(0);
+		for (PowerSet ps : engagedSet.getOnPowerSet()) {
+			// Equivalent means that they have the same number of subassemblies
+			Assembly pa = new Assembly(crystalAssemblies, ps);
+			if (pa.getAssemblyGraph().getSubAssemblies().size() == 
+					this.getAssemblyGraph().getSubAssemblies().size())
+				pss.add(ps);
+		}
+		pss.add(engagedSet);
 		
-		setCall(firstSubAssembly.score());
-				
+		// Calculate the probabilities for each possible set of interface clusters
+		probability = 0;
+		InterfaceEvolContextList iecl = getCrystalAssemblies().getInterfaceEvolContextList();
+					
+		for (PowerSet ps : pss) {
+			double prob = 1;
+			for (int i = 1; i < ps.size()+1; i++) {
+				double p = iecl.getCombinedClusterPredictor(i).getScore();
```

Comment:
True, this was one of my questions in #147. If you have the right way to do it, we can update the code.